Dockerfile

```diff
@@ -1,11 +1,20 @@
 FROM dmscid/lewis-depends:latest
 
+# Copying this separately from the rest of lewis allows the
+# pip install step to be cached until the requirements change.
+# Rebuilding is faster and doesn't require an internet connection.
+COPY requirements.txt /lewis/
+
+# Install any missing PIP requirements, clear PIP cache, and delete all compiled Python objects
+RUN pip install -r /lewis/requirements.txt && \
+    rm -rf /root/.cache/pip/* && \
+    find \( -name '*.pyc' -o -name '*.pyo' \) -exec rm -rf '{}' +
+
 # Lewis source is pulled in from current directory (note .dockerignore filters out some files)
 COPY . /lewis
 
-# Install any missing PIP requirements, clear PIP cache, and delete all compiled Python objects
+# Install lewis in-place and remove compiled Python objects
 RUN pip install -e /lewis && \
```

Comment:
I see... only reason I brought up the imports is because `$ python src/lewis` throws an exception on that import (without lewis installed as a package).

Anyway, I'm happy to keep `pip install -e`... but then we should probably also have a `rm -rf /root/.cache/pip/* &&` step after it, same as after the `pip install -r`, just in case pip decides to stuff something in there.